### PR TITLE
Only call DisconnectNodes once per second

### PR DESCRIPTION
**Pull request by codablock**
See individual commits. This PR also includes "Only run InactivityCheck once per second" from #3420 to avoid future merge conflicts. I'll rebase this PR after #3420 is merged into develop.

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
@codablock tests failing

File: src/net.cpp [hunk 1]
```diff
@@ -1660,16 +1661,23 @@ void CConnman::SocketHandler()
...
 {
+    int64_t nLastCleanupNodes = 0;
+
     while (!interruptNet)
     {
-        DisconnectNodes();
+        if (GetTimeMillis() - nLastCleanupNodes > 1000) {
+            ForEachNode(AllNodes, [&](CNode* pnode) {
+                InactivityCheck(pnode);
+                DisconnectNodes();
```
Comment (UdjinM6) [inline src/net.cpp:1677]: This should be called outside of `ForEachNode`

Comment (codablock) [reply src/net.cpp:1677]: Oh, right. Fixed.

**Comment by codablock** [comment]
Tests are hopefully fixed now (had to call DeleteNode() without any locks held)

**Comment by UdjinM6** [comment]
`vNodes`/`vNodesDisconnected` are also accessed to delete nodes in `CConnman::Stop()`. Shouldn't we protect them there now that we call `DisconnectNodes()` not only in `ThreadSocketHandler` but in many other places? e.g. smth like 3ae02dd42b

**Comment by codablock** [comment]
@UdjinM6 yeah I noticed this as well, but decided to not fix it and leave it the way it is. Bitcoin also never fixed the missing cs_vNodes lock. It's semi-ok in this case, as Stop() first ensures that all network threads are stopped, so there is nothing left that could access the vectors.

**Comment by UdjinM6** [comment]
> Stop() first ensures that all network threads are stopped, so there is nothing left that could access the vectors.

Well, we call `DisconnectNodes()` from various rpc commands also now but as far as I can see RPC is stopped before connman so yeah, should be ok probably but looks a bit scary :)

**Pull request merged**